src/e2e/scala/ie/zalando/fabric/gateway/spec/CorsSpec.scala

```diff
@@ -0,0 +1,61 @@
+package ie.zalando.fabric.gateway.spec
+
+import com.softwaremill.sttp._
+import ie.zalando.fabric.gateway.TestConstants._
+import ie.zalando.fabric.gateway.{LoggingSttpBackend, TestConstants}
+import org.scalatest._
+
+class CorsSpec extends FunSpec with Matchers {
+
+  implicit val backend = new LoggingSttpBackend[Id, Nothing](HttpURLConnectionBackend())
+
+  describe("Fabric Gateway Cors Support") {
+    it("should allow unauthenticated options requests when cors is configured") {
+      val origin1 = sttp
+        .options(TestConstants.TestAppResourceById(123))
+        .header("Origin", "https://example.com")
+        .send()
+
+      val origin2 = sttp
+        .options(TestConstants.TestAppResourceById(123))
+        .header("Origin", "https://example-other.com")
+        .send()
+
+      val notAllowed = sttp
+        .options(TestConstants.TestAppResourceById(123))
+        .header("Origin", "https://example-not-allowed.com")
+        .send()
+
+      origin1.code shouldBe 204
+      origin2.code shouldBe 204
+      notAllowed.code shouldBe 204
+
+      origin1.headers should contain allOf (
+        "Access-Control-Allow-Methods"     -> "GET, PUT, OPTIONS",
+        "Access-Control-Allow-Credentials" -> "true",
+        "Access-Control-Allow-Headers"     -> "X-Flow-Id, Content-Type",
+        "Access-Control-Allow-Origin"      -> "https://example.com",
+      )
+      origin2.headers should contain allOf (
+        "Access-Control-Allow-Methods"     -> "GET, PUT, OPTIONS",
+        "Access-Control-Allow-Credentials" -> "true",
+        "Access-Control-Allow-Headers"     -> "X-Flow-Id, Content-Type",
+        "Access-Control-Allow-Origin"      -> "https://example-other.com",
+      )
+      notAllowed.headers should contain allOf (
+        "Access-Control-Allow-Methods"     -> "GET, PUT, OPTIONS",
+        "Access-Control-Allow-Credentials" -> "true",
```

Comment:
Should we be returning allow creds here? What is the significance of it for a request from an unsupported origin?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials  `When used as part of a response to a preflight request, this indicates whether or not the actual request can be made using credentials`. If there is an unsupported origin the preflight won't include the origin in the response, so the browser won't make the subsequent request. 

The situation with allow credentials is not ideal in general, we only want to return cors headers if the request has an Origin header. But except for the allowed-origin header (from the skipper cors filter) we can only append them to all requests. If we want to go further we would need to make a cors filter for skipper that supports more config options.